### PR TITLE
[bug](function)fix json_replace check return type error

### PR DESCRIPTION
**Pull request by zhangstar333**
## Proposed changes
1. fix the return type dcheck error:
```
mysql [test]>select (json_replace(a, '$.fparam.nested_2', "qwe")) from json_table_2 limit 1;
ERROR 1105 (HY000): errCode = 2, detailMessage = (10.16.10.8)[INTERNAL_ERROR]Function json_replace get failed, expr is VectorizedFnCall[json_replace](arguments=a, String, String, String,return=Nullable(String)) and return type is Nullable(String).
```

2.  improve the json_replace/json_insert/json_set function execute of not convert const column, test about could faster 1s on 1000w table rows

### CODE REVIEW

**Comment by zhangstar333** [comment]
run buildall

File: be/src/vec/functions/function_json.cpp [hunk 1]
```diff
@@ -1329,34 +1331,38 @@ class FunctionJsonModifyImpl : public IFunction {
...
+        if (is_nullable) {
+            ret_null_map = ColumnUInt8::create(input_rows_count, 0);
+        }
+        RETURN_IF_ERROR(
+                execute_process(data_columns, *assert_cast<ColumnString*>(result_column.get()),
+                                input_rows_count, nullmaps, is_nullable,
+                                *assert_cast<ColumnUInt8*>(ret_null_map.get()), column_is_consts));
 
         if (is_nullable) {
             block.replace_by_position(result, ColumnNullable::create(std::move(result_column),
```
Comment (HappenLee) [inline be/src/vec/functions/function_json.cpp:1368]: do not create `ret_null_map` before really need the column

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by wangbo — APPROVED** [review]
LGTM

**Pull request merged**